config/management/genesis/src/fullnode_builder.rs

```diff
@@ -84,6 +109,58 @@ impl FullnodeConfig {
         }
     }
 
+    fn attach_to_validator(&mut self, validator_config: &mut NodeConfig) -> Result<()> {
+        ensure!(
+            matches!(validator_config.base.role, RoleType::Validator),
+            "Validator config must be a Validator config"
+        );
+
+        // Grab the public network config from the validator and insert it into the VFN's config
+        let public_network = {
+            let (i, _) = validator_config
+                .full_node_networks
+                .iter()
+                .enumerate()
+                .find(|(_i, config)| config.network_id == NetworkId::Public)
+                .expect("Validator should have a public network");
+            validator_config.full_node_networks.remove(i)
+        };
+
+        let fullnode_public_network = self
+            .config
+            .full_node_networks
+            .iter_mut()
+            .find(|config| config.network_id == NetworkId::Public)
+            .expect("VFN should have a public network");
+        fullnode_public_network.identity = public_network.identity;
+        fullnode_public_network.listen_address = public_network.listen_address;
```

Comment:
Will this have any conflicts between the two?

Reply:
no because it was removed from the Validator's config, like you identified above (and then the caller is expected to re-save and restart the validator)